llms/mlx_lm/utils.py

```diff
@@ -113,11 +138,21 @@ def sample(logits: mx.array) -> Tuple[mx.array, float]:
         prob = softmax_logits[0, token]
         return token, prob
 
+    if repetition_penalty < 0 or not isinstance(repetition_penalty, float):
+        raise ValueError(
+            f"repetition_penalty must be a non-negative float, got {repetition_penalty}"
+        )
+
     y = prompt
     cache = None
+
     while True:
-        logits, cache = model(y[None], cache=cache)
-        logits = logits[:, -1, :]
+        logits, cache = model(y[:, None], cache=cache)
```

Comment:
Issue: The batch dimensions should be the first dimensions. Please revert this back to `y[None]` and revert `logits = logits.squeeze(1)` to `logits = logits[:, -1, :]`, as we need to retrieve the last token from the sequence dimension instead of removing sequence dimensions.